_test/shared/noisify_test_rand.m

```diff
@@ -0,0 +1,65 @@
+classdef noisify_test_rand < handle
+%--------------------------------------------------
+% This class creates a simple deterministic series
+% of numbers which can be substituted for a true pseudo-random
+% number generator to enable by-eye checking of the random
+% noise addition (via noisify( to the signal. It can
+% substitute for randn in the Herbert noisify method.
+%
+% This function is designed to accumulate the "pseudo-
+% random" numbers across multiple calls - one for each
+% page of a paged sqw object. An external global flag
+% initialises these numbers, and an internal persistent
+% number keeps track of where this process has got to.
+%
+% Input to the generator function myrand:
+%     n - size of the signal vector for which the noise
+%         is being generated.
+% Where multiple tests are required a separate class object
+% can be used for each one.
+%--------------------------------------------------
+
+    properties
+        % flag to signal that this pseudo-random
+        % number distribution needs to be initialised if true.
+        init = 1
+        % number tracking where in the generation of
+        % the sequence we have got to. Initially a dummy bad value.
+        got_to = -666
+    end
+    methods
+        function vals = myrand(obj,size)
+            % On initialisation the flag is reset false and the number
+            % sequence tracker is initialised to 1
+            if obj.init==1
+                obj.init=0;
+                obj.got_to=1;
+            end
+            
+            % the size vector of the signal which is passed in as the argument
+            % is a 1xsize vector - the actual size is extracted from the second 
```

Comment:
this comment is describing how MATLAB works.

Reply:
No, it's describing how the expected argument will function. It is possible for a user to input a simple scalar - which would be wrong - this interface is expecting size(y) where y is the signal vector.

Reply:
I understand.

This distinction would be clearer if rather than reassigning the `size` variable here a new variable, e.g. "num_elements" or "signal_length" was created.

Reply:
That is useful. Harry comments that size being a matlab function it should not be reused as a variable name - fixing that will cover this.